src/Type/IntegerRangeType.php

```diff
@@ -281,11 +291,11 @@ private function isSubTypeOfUnion(UnionType $otherType): TrinaryLogic
 			);
 
 			if (count($matchingConstantIntegers) === ($this->max - $this->min + 1)) {
-				return TrinaryLogic::createYes();
+				return AcceptsResult::createYes();
 			}
 		}
 
-		return TrinaryLogic::createNo()->lazyOr($otherType->getTypes(), fn (Type $innerType) => $this->isSubTypeOf($innerType));
+		return AcceptsResult::createNo()->or(...array_map(fn (Type $innerType) => $this->isSubTypeOfWithReason($innerType), $otherType->getTypes()));
```

Comment:
Here we're loosing a `lazyOr` into a `or`.

I dunno if the lazy could be kept on AcceptsResults but I saw similar lazy lost when `isAcceptedWithReasonBy` was introduced